src/data_readers/data_reader.cpp

```diff
@@ -58,6 +58,13 @@ void generic_data_reader::setup() {
   set_initial_position();
 
   shuffle_indices();
+
+  m_thread_buffer.resize(omp_get_max_threads(), std::vector<char>());
+#pragma omp parallel
```

Comment:
It seems simpler to loop through the buffers rather than launching an OpenMP section.

Reply:
Good point.